sdk/communication/communication-identity/test/public/utils/matrix.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ * @internal
+ * Takes a jagged 2D array and a function and runs the function with every
+ * possible combination of elements of each of the arrays
+ *
+ * For strong type-checking, it is important that the `matrix` have a strong
+ * type, such as a `const` literal.
+ *
+ * @param values - jagged 2D array specifying the arguments and their possible
+ *                 values
+ * @param handler - the function to run with the different argument combinations
+ *
+ * @example
+ * ```typescript
+ * matrix([
+ *     [true, false],
+ *     [1, 2, 3]
+ *   ] as const,
+ *   (useLabels: boolean, attempts: number) => {
+ *     // This body will run six times with the following parameters:
+ *     // - true, 1
+ *     // - true, 2
+ *     // - true, 3
+ *     // - false, 1
+ *     // - false, 2
+ *     // - false, 3
+ *   });
+ * ```
+ */
+export function matrix<T extends ReadonlyArray<readonly unknown[]>>(
```

Comment:
This matrix func is being used in a few places already under communication, under phone number and sms. 
Does it now make sense to move this into a common place?

Reply:
I think the consensus is we should, but for this story we're going to put this here for now.

Reply:
@JoshuaLai looks like @0rland0Wats0n has a story to do this this sprint :)